libpldmresponder/fru.cpp

```diff
@@ -341,6 +375,393 @@ void FruImpl::populateRecords(
+            std::move(std::vector<uint8_t>(1, PLDM_RECORDS_ADDED)));
+    }
+    if (!handlesTobeModified.empty())
+    {
+        sendPDRRepositoryChgEventbyPDRHandles(
+            std::move(handlesTobeModified),
+            std::move(std::vector<uint8_t>(1, PLDM_RECORDS_MODIFIED)));
+    }
+}
+
+void FruImpl::reGenerateStatePDR(const std::string& fruObjectPath,
+                                 std::vector<uint32_t>& recordHdlList)
+{
+    pldm::responder::pdr_utils::Type pdrType{};
+    static const Json empty{};
+    for (const auto& directory : statePDRJsonsDir)
+    {
+        for (const auto& dirEntry : fs::directory_iterator(directory))
+        {
+            try
+            {
+                if (fs::is_regular_file(dirEntry.path().string()))
+                {
+                    auto json = pldm::responder::pdr_utils::readJson(
+                        dirEntry.path().string());
+                    if (!json.empty())
+                    {
+                        pldm::responder::pdr_utils::DbusObjMaps tmpMap{};
+                        auto effecterPDRs = json.value("effecterPDRs", empty);
+                        for (const auto& effecter : effecterPDRs)
+                        {
+                            pdrType = effecter.value("pdrType", 0);
+                            if (pdrType == PLDM_STATE_EFFECTER_PDR)
+                            {
+                                auto stateEffecterList = setStatePDRParams(
+                                    {directory}, 0, 0, tmpMap, tmpMap, true,
+                                    effecter, fruObjectPath, pdrType);
+                                std::move(stateEffecterList.begin(),
+                                          stateEffecterList.end(),
+                                          std::back_inserter(recordHdlList));
+                            }
+                        }
+                        auto sensorPDRs = json.value("sensorPDRs", empty);
+                        for (const auto& sensor : sensorPDRs)
+                        {
+                            pdrType = sensor.value("pdrType", 0);
+                            if (pdrType == PLDM_STATE_SENSOR_PDR)
+                            {
+                                auto stateSensorList = setStatePDRParams(
+                                    {directory}, 0, 0, tmpMap, tmpMap, true,
+                                    sensor, fruObjectPath, pdrType);
+                                std::move(stateSensorList.begin(),
+                                          stateSensorList.end(),
+                                          std::back_inserter(recordHdlList));
+                            }
+                        }
+                    }
+                }
+            }
+            catch (const InternalFailure& e)
+            {
+                error(
+                    "PDR config directory does not exist or empty, TYPE= {PDR_TYP} PATH= {DIR_PATH} ERROR={ERR_EXCEP}",
```

Comment:
please maintain trace format all over the patch 
FYR: https://gerrit.openbmc.org/c/openbmc/pldm/+/70327
https://gerrit.openbmc.org/c/openbmc/pldm/+/70559

Reply:
+1